adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -396,9 +353,9 @@ final void handleEndpointConnection(final MqttEndpoint endpoint) {
 
     private Future<Device> handleConnectionRequest(final MqttEndpoint endpoint, final Span currentSpan) {
 
-        if (isConnectionLimitExceeded()) {
-            LOG.debug("Connection limit ({}) exceeded, reject connection request", connectionLimit);
-            currentSpan.log(String.format("connection limit (%d) exceeded", connectionLimit));
+        // the ConnectionLimitManager is null in some unit tests
```

Comment:
would it make sense to initialize the connectionLimitManager as part of the constructor?

Reply:
The Constructor of the protocol adapter? The ConnectionLimitManager can be provided by a setter and depends on properties that are not present during construction as well (metrics, config), so I don't see how this could be done in the constructor.  The other option I see would be to leverage one of Spring's initialization options. If the goal is to prevent  connectionLimitManager  from being null, then it could be provided as a Bean and initialized with a NoOp implementation or Optional. Not sure if this would be more obvious.

Reply:
oh sure, I forgot about the properties that it depends on  ...